dspace-server-webapp/src/main/java/org/dspace/app/rest/utils/SolrOAIReindexer.java

```diff
@@ -309,44 +310,182 @@ public void reindexItem(Item item) {
 //            we cannot put such item into solr
             return;
         }
+
+        // Check for test property to force event-based fallback
+        if ("true".equals(System.getProperty("dspace.test.force.event.fallback", "false"))) {
```

Comment:
1. What.. use getBooleanProperty and compare it as boolean. Can't you use ConfigurationSErvice?
2. Where is defined that property?

Reply:
It is not a DSpace configuration property, so you cannot use ConfigurationService for it. It is used for testing. It is not defined by default. When it is not defined, its default value is false. However, in the test environment, we set it to true, so it exists only during testing.